06_week/game.js

```diff
@@ -0,0 +1,120 @@
+let Player = { x: 0, y: 0 };
+let restartButton = document.getElementById("restart-button");
+
+document.addEventListener("keydown", function (event) {
+  keydownListener(event.key);
+});
+
+restartButton.addEventListener("click", (event) => {
+  playground.parentElement.removeChild(playground);
+  createPlayground();
+  let numberOfObstacles = Math.floor(Math.random() * 30 + 10);
+  console.log(numberOfObstacles);
+  for (i = 0; i < numberOfObstacles; i++) {
+    createObstacles();
+  }
+});
+
+function createPlayground() {
+  // 1. cream playground-ul
+  let playground = document.createElement("div");
+  playground.setAttribute("id", "game-container");
+
+  // 2. cream playerul cu o pozitie random(?) si il adaugam in playground
+
+  // let posX = Math.floor(Math.random() * 980);
+  // let posY = Math.floor(Math.random() * 480);
+  let ourPlayer = document.createElement("div");
+  ourPlayer.setAttribute("id", "player");
+  // ourPlayer.style.left = `${posX}px`
+  // ourPlayer.style.top = `${posY}px`
+  // Player.x = posX;
+  // Player.y = posY;
+  playground.appendChild(ourPlayer);
+
+  document.body.appendChild(playground);
+}
+
+let numberOfObstacles = Math.floor(Math.random() * 30 + 10);
+console.log(numberOfObstacles);
+for (i = 0; i < numberOfObstacles; i++) {
+  createObstacles();
+}
+
+// 3. cream numar random de obstacole , cu pozitii si dimensiuni random si le introducem pe playground
+function createObstacles() {
+  let width = Math.floor(Math.random() * 70 + 10);
+  let height = Math.floor(Math.random() * 70 + 10);
+  let obstaclePosX = Math.floor(Math.random() * 980);
+  let obstaclePosY = Math.floor(Math.random() * 480);
+
+  let obstacle = document.createElement("div");
+  obstacle.setAttribute("class", "obstacle");
+  obstacle.style.left = `${obstaclePosX}px`;
+  obstacle.style.top = `${obstaclePosY}px`;
+  obstacle.style.width = `${width}px`;
+  obstacle.style.height = `${height}px`;
+  if (isObstacleInsideDiv(obstaclePosX, obstaclePosY, height, width)) {
+    playground.appendChild(obstacle);
+  } else {
+    console.log("Obstacle is spawning outside the playground , skipping ... ");
+  }
+}
+
+function isPlayerInsideDiv(x, y) {
+  if (x < 0 || playground.offsetWidth - 20 < x) {
+    return false;
+  }
+  if (y < 0 || playground.offsetHeight - 20 < y) {
+    return false;
+  }
+  return true;
+}
+
+function isObstacleInsideDiv(x, y, height, width) {
+  if (x < 0 || playground.offsetWidth - width < x) {
+    return false;
+  }
+  if (y < 0 || playground.offsetHeight - height < y) {
+    return false;
+  }
+  return true;
+}
+function keydownListener(key) {
+  console.log(key);
+  let x = Player.x;
+  let y = Player.y;
+  switch (key) {
+    case "ArrowUp":
+      y -= 20;
+      break;
+    case "ArrowLeft":
+      x -= 20;
+      break;
+    case "ArrowDown":
+      y += 20;
+      break;
+    case "ArrowRight":
+      x += 20;
+      break;
+    default:
+      console.log("Input is not valid");
+  }
+  movePlayer(x, y);
```

Comment:
super

Reply:
aici in loc sa pui apelu de movePlayer aici il poti muta. Iar keydownListener doar returneaza {x, y}